main.go

```diff
@@ -70,13 +73,34 @@ func setupEnvironment() {
 	viper.SetDefault("IGNORE_FILE_REGEX", `README.md`)
 	viper.SetDefault("VERBOSE", "false")
 	viper.AutomaticEnv()
-	viper.BindEnv("CONSUL_KEY_PREFIX")
-	viper.BindEnv("DEFAULT_CONFIG_TYPE")
-	viper.BindEnv("DIRECTORY")
-	viper.BindEnv("DRYRUN")
-	viper.BindEnv("IGNORE_DIR_REGEX")
-	viper.BindEnv("IGNORE_FILE_REGEX")
-	viper.BindEnv("VERBOSE")
+	err := viper.BindEnv("CONSUL_KEY_PREFIX")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = viper.BindEnv("DEFAULT_CONFIG_TYPE")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = viper.BindEnv("DIRECTORY")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = viper.BindEnv("DRYRUN")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = viper.BindEnv("IGNORE_DIR_REGEX")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = viper.BindEnv("IGNORE_FILE_REGEX")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = viper.BindEnv("VERBOSE")
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
I don't love this approach, as if there are multiple errors you will only see them one at a time as log.Fatal will exit.  Plus, since the logic is the same everywhere.   Personally I'd prefer a loop with list of env vars you want to bind.

Reply:
I've replaced the repetition with loops. As for failing one at a time, I'm ok with that given the error condition being checked is the input having 0 length. That only happens if we screw up the contents of the envDefaults map.